packages/gestalt/src/BannerOverlay.js

```diff
@@ -0,0 +1,262 @@
+      }
+    | {
+        accessibilityLabel: string,
+        label: string,
+        onClick: $ElementType<ElementConfig<typeof Button>, 'onClick'>,
+        role?: 'button',
+        size?: $ElementType<ElementConfig<typeof Button>, 'size'>,
+      },
+  /**
+   * An optional thumbnail to display next to the text.
+   */
+  thumbnail?:
+    | { image: Element<typeof Image> }
+    | { avatar: Element<typeof Avatar> }
+    | { icon: Element<typeof Icon> },
+  /**
+   *  Heading of BannerOverlay. Content should be [localized](https://gestalt.pinterest.systems/web/banneroverlay#Localization). See the [Text variant](https://gestalt.pinterest.systems/web/banneroverlay#Text) to learn more.
+   */
+  title?: string,
+  /**
+   *  zIndex of BannerOverlay. See the [zIndex guidelines](https://gestalt.pinterest.systems/web/banneroverlay#zIndex) to learn more.
+   */
+  zIndex?: Indexable,
+};
+
+/**
+ * [BannerOverlays](https://gestalt.pinterest.systems/web/banneroverlay) displays short educational messages when users have performed actions that indicate some intent, such as a related pin tap or idea pin swipe. It is scrim-less, meaning users can scroll content underneath without having a wash layer on top of the content.
+ *
+ * BannerOverlay is a sticky component triggered by scroll, on tap or long-press, placed at the bottom of the screen.
+ * ![BannerOverlay light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/BannerOverlay.spec.mjs-snapshots/BannerOverlay-chromium-darwin.png)
+ */
+export default function BannerOverlay({
+  message,
+  title,
+  onDismiss,
+  helperLink,
+  primaryAction,
+  offset = { top: 0, bottom: 0 },
+  thumbnail,
+  zIndex,
+}: Props): ReactNode {
+  const { name: colorSchemeName } = useColorScheme();
+  const isDarkMode = colorSchemeName === 'darkMode';
+
+  const isMobileDevice = useDeviceType() === 'mobile';
+
+  let messageTextElement: Element<'span'> | string;
+
+  if (typeof message === 'string') {
+    messageTextElement = message;
+  }
+  if (typeof title === 'string') {
+    messageTextElement = title;
+  }
+
+  // If `message` is a Text component, we need to override any text colors within to ensure they all match
+  const checkTextNode = () => {
+    const messageIsTextNode =
+      typeof message !== 'string' &&
+      Children.only<Element<typeof Text>>(message).type.displayName === 'Text';
+
+    if (messageIsTextNode) {
+      const textColorOverrideStyles = isDarkMode
+        ? styles.textColorOverrideDark
+        : styles.textColorOverrideLight;
+
+      messageTextElement = <span className={textColorOverrideStyles}>{message}</span>;
+    }
+    return messageIsTextNode;
+  };
+
+  const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
+    useDefaultLabelContext('BannerOverlay');
+
+  const { lightModeBackground, darkModeBackground, textColor } = DEFAULT_COLORS;
+
+  const dismissButtonComponent = !!onDismiss && (
```

Comment:
Keep the conditional in the main logic for readability, because it's hard to follow if this will render or not

so {!!onDismiss && dismissButtonComponent}

Reply:
Done!